### ISSUE TITLE
again on no-resign games:  last game took 421063 seconds :-( 

### ISSUE THREAD

**Issue by maxinjapan** [gcp/leela-zero#1789]
Hi all, 
I just noticed that [the last self-play game on my pc took 421063 seconds](http://zero.sjeng.org/view/d8a9222bb1c36e87afb7f14fb639270ee801fee75ebcdabab0046b427a61f91f?viewer=wgo)! 

Stats before that game:
> 138 game(s) (102 self played and 36 matches) played in 2920 minutes = 1269 seconds/game, 5707 ms/move, last game took 619 seconds.

Stats after that game: 
> 139 game(s) (103 self played and 36 matches) played in 9938 minutes = 4289 seconds/game, 19030 ms/move, last game took 421063 seconds.

Any suggestion about why this has happened? My pc is set never to sleep, but the screen times out after 15 minutes

**Comment by l1t1** [comment]
maybe data was overflow

**Comment by maxinjapan** [comment]
hi @l1t1 how do I do that?

**Comment by diadorak** [comment]
Were you using the GPU to do something else at the same time?

**Comment by sheeryjay** [comment]
@maxinjapan did you perhaps put your PC to sleep or hibernate? Your numbers would imply that said no-resign game took about 400x more time than one normal (or even some no-resign) before that. It must be a bug in time calculation.

**Comment by maxinjapan** [comment]
@diadorak no the PC was left alone for the entire time.

@sheeryjay this was my guess too, but I cannot see any settings on my pc allowing hybernation or sleep. 

The game did take 4+ days as it was the only game completed since the previous time I used the PC on Saturday morning. I do not know what's happened in the meanwhile.

**Comment by l1t1** [comment]
View the historical boot/shutdown time of Windows 7
[Computer] ⇒ (Right click) [Manage] ⇒ [Event Viewer] ⇒ [Windows Log] ⇒ [System]

On the right side [Filter Current Log], in the pop-up dialog box, the event source selects Kernel-Power (indicating power events, power-on, power-off, hibernation, etc.), and the task types are as follows:

     86: Historical boot time
     103: Historical shutdown time

Click OK and it will list all the boot and shutdown times.

**Comment by ajcsimons** [comment]
During normal games each player is making sensible moves that the other probably predicted as a response after their move so they can reuse a bunch of visits already in the game tree; whereas with these no-resign games of random nonsense it's much more unlikely the random fill-in-own-eyes move of the opponent was predicted so every move is a full regenerate of the game tree with 1601 visits and takes longer. At least that was my conclusion when normal games take 10 minutes for me and no-resign ones 3 hours. Incidentally I find that rather demotivating to contribute my GPU time/electricity.

**Comment by Ishinoshita** [comment]
@ajcsimons I once estimated that moves beyond resignation threshold account for 25% of the total positions in the training samples *and of the total computing ressources of the project*.

Your point on tree reuse makes clear that the later is obviously underestimated. I failed to factor the drop in the tree reuse when the win rate collapses and the policy flattens (typically, policy max drops below 20% on average, meaning that moves picked by temperature can be far in the move rank and that they typically will have as few as 5-10% of visits on average. I used move as the unit to estimate the computation cost, which was wrong. 

Making *very coarse* assumptions based on your numbers. Based on 10' vs 3h, I estimated a drop of a factor 8 in tree reuse for the moves past resignation threshold.
(assuming resign games of 165 moves, no-resign games of 525 moves on average). This sounds very plausible in view of my own previous data. 

Assuming this 8 factor, I computed a shocking 70% of project ressources used for generating moves past resignation threshold, in the 15% of self-play games that are no-resign games. Of course, that is no stats! Just a mere indication of how surprising the real stats might be.

Let someone else redo the estimation properly and the math, if the subject is deemed of importance. E.g. at server side, where client can easily be identified (and assuming a client uses a constant hardware) it might be feasible to evaluate the ratio between resign and no-resign games in terms of moves/minute, from which it could be possible to estimate the tree reuse drop factor.

*I don't suggest there is any issue with this use of the ressources, I just note the recurrent questioning/frustration in the late threads in this regard, that may need extra attention, explanations, to keep contributors good spirited.*

* It is no dispute that LZ is a wonderful project and already an amazing achievement. I'm personally considering upgrading my PC with a decent GPU to contribute really (I symbolically contributed few 10s of games with my old CPU only hardware but stopped).*
